### PR TITLE
Document trigger policies for modules

### PR DESCRIPTION
**Pull request by marcinwyszynski**
## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/policy/trigger-policy.md [hunk 1]
```diff
@@ -114,6 +114,42 @@ This is the schema of the data input that each policy request will receive:
 !!! info
     Note the presence of two similar keys: `stack` and `stacks`. The former is the Stack that the newly finished Run belongs to. The other is a list of all Stacks in the account. The schema for both is the same.
 
+When triggered by a _new module version_, this is the schema of the data input that each policy request will receive:
+
+```json
+{
+  "module": { // Module for which the new version was released
```
Comment (adamconnelly) [inline docs/concepts/policy/trigger-policy.md:121]: I think the module object can also contain `project_root` and `contexts`.

File: docs/concepts/policy/trigger-policy.md [hunk 2]
```diff
@@ -114,6 +114,42 @@ This is the schema of the data input that each policy request will receive:
...
+    "labels": ["string - list of arbitrary, user-defined selectors"],
+    "namespace": "string - repository namespace, only relevant to GitLab repositories",
+    "repository": "string - name of the source repository",
+    "terraform_provider": "string - name of the main Terraform provider used by the module",
+    "version": { // Newly released module version
+      "number": "string - semver version number",
+      "created_at": "number - creation Unix timestamp in nanoseconds",
+    }
+  }
+  "stacks": [ // List of consumers of the newest available module version
```
Comment (adamconnelly) [inline docs/concepts/policy/trigger-policy.md:134]: It looks like `stacks` can also contain `tracked_commit`, `worker_pool` and `contexts`.

File: docs/concepts/policy/trigger-policy.md [hunk 3]
```diff
@@ -275,3 +311,7 @@ trigger[stack.id] {
   count(finished_dependencies) == count(dependencies)
 }
 ```
+
+### Module updates
+
+Trigger policies can be attached to modules as well. Modules track the consumers of each of their versions. When a new module version is released, the consumers of the previously newest version are assumed to be potential consumers of the newly released one. Hence, the trigger policy for a module can be used to trigger a run on all of these stacks. The module version will be updated as long as the version constraints allow the newest version to be used.
```
Comment (adamconnelly) [inline docs/concepts/policy/trigger-policy.md:317]: Maybe worth including an example policy as well?

**Review by adamconnelly — COMMENTED** [review]
LGTM - just some minor suggestions.

**Review by adamconnelly — COMMENTED** [review]
LGTM - just some minor suggestions.

File: docs/concepts/policy/trigger-policy.md (hunk 1 above)
Comment (marcinwyszynski) [reply docs/concepts/policy/trigger-policy.md:121]: Good point about the project root. We don't load contexts for this policy though.

File: docs/concepts/policy/trigger-policy.md (hunk 3 above)
Comment (marcinwyszynski) [reply docs/concepts/policy/trigger-policy.md:357]: Sure, done.

File: docs/concepts/policy/trigger-policy.md (hunk 2 above)
Comment (marcinwyszynski) [reply docs/concepts/policy/trigger-policy.md:161]: See the note above re: contexts. Added the rest.

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**